### PR TITLE
Update facebook-ios-sdk to 4.15.1

### PR DESCRIPTION
**Pull request by spacelatte**
#### Checklist
- [x] The commit message includes the cask’s name and version.
- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` left no offenses.

### CODE REVIEW

**Comment by vitorgalvao** [comment]
Polluting a user’s `Documents` directory seems wrong. Is there any details about this installation and why it changed?

File: Casks/facebook-ios-sdk.rb [hunk 1]
```diff
@@ -1,13 +1,22 @@
 cask 'facebook-ios-sdk' do
-  version '4.14.0'
-  sha256 '1da24bf34eae840f6eed22912b3d2818d79bfc171b7269d554c8a28ed86257bf'
+  version '4.15.1'
+  sha256 '255b0b47cbad187dbb8834ed49e917106fb1d38815a8d88d57c816ac741a4ed5'
 
-  url "https://origincache.facebook.com/developers/resources/?id=facebook-ios-sdk-#{version}.pkg"
+  url 'https://origincache.facebook.com/developers/resources/?id=facebook-ios-sdk-current.zip'
```
Comment (vitorgalvao) [inline Casks/facebook-ios-sdk.rb]: Is there no versioned URL anymore?

Comment (pvtmert) [inline Casks/facebook-ios-sdk.rb]: it seems no, also facebook recommends to put it there, i am updating cask because of yesterdays' discussion on gitter

Comment (pvtmert) [inline Casks/facebook-ios-sdk.rb]: btw, if i say `#{ENV['HOME']}/Documents/FacebookSDK` will it create that dir or fails,
according to that i can update

Comment (vitorgalvao) [inline Casks/facebook-ios-sdk.rb]: It should fail. Use [`kicad`](https://github.com/caskroom/homebrew-cask/blob/7b96fd2831f381b8e9ec07b980a32788c269b2d2/Casks/kicad.rb) as an example  to deal with that.

File: Casks/facebook-ios-sdk.rb [hunk 2]
```diff
@@ -1,13 +1,22 @@
 cask 'facebook-ios-sdk' do
-  version '4.14.0'
-  sha256 '1da24bf34eae840f6eed22912b3d2818d79bfc171b7269d554c8a28ed86257bf'
+  version '4.15.1'
```
Comment (vitorgalvao) [inline Casks/facebook-ios-sdk.rb]: Seeing as `url` doesn’t point to a specific version, this cask should use `version :latest` and `sha256 :no_check`. [See an example](https://github.com/caskroom/homebrew-cask/blob/f25b6babcd398abf48e33af3d887b2d00de1d661/Casks/kibako.rb).

**Comment by fanquake** [comment]
```
==> 1 modified cask: facebook-ios-sdk
audit for facebook-ios-sdk: passed
>>> brew cask style Casks/facebook-ios-sdk.rb
== Casks/facebook-ios-sdk.rb ==
C: 22:  1: stanza groups should be separated by a single empty line
C: 22:  1: Trailing whitespace detected.
1 file inspected, 2 offenses detected
Error: style check failed
```

File: Casks/facebook-ios-sdk.rb [hunk 3]
```diff
@@ -1,13 +1,26 @@
...
+  sha256 :no_check
 
-  url "https://origincache.facebook.com/developers/resources/?id=facebook-ios-sdk-#{version}.pkg"
+  url 'https://origincache.facebook.com/developers/resources/?id=facebook-ios-sdk-current.zip'
   name 'Facebook SDK for iOS'
   homepage 'https://developers.facebook.com/docs/ios'
   license :unknown # TODO: change license and remove this comment; ':unknown' is a machine-generated placeholder
 
-  pkg "facebook-ios-sdk-#{version}.pkg"
+  artifact 'AccountKit.framework', target: "#{ENV['HOME']}/Documents/FacebookSDK"
```
Comment (vitorgalvao) [inline Casks/facebook-ios-sdk.rb]: You need to add the artifact’s name at the end, as well (`#{ENV['HOME']}/Documents/FacebookSDK/AccountKit.framework`).

Also test `brew cask install`, to make sure we’re not forgetting anything else.

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

File: Casks/facebook-ios-sdk.rb [hunk 4]
```diff
@@ -1,13 +1,27 @@
...
+  sha256 :no_check
 
-  url "https://origincache.facebook.com/developers/resources/?id=facebook-ios-sdk-#{version}.pkg"
+  url 'https://origincache.facebook.com/developers/resources/?id=facebook-ios-sdk-current.zip'
   name 'Facebook SDK for iOS'
   homepage 'https://developers.facebook.com/docs/ios'
   license :unknown # TODO: change license and remove this comment; ':unknown' is a machine-generated placeholder
 
-  pkg "facebook-ios-sdk-#{version}.pkg"
+  preflight do
```
Comment (vitorgalvao) [inline Casks/facebook-ios-sdk.rb]: Stanza order does not affect execution order, but it is [important for style](https://github.com/caskroom/homebrew-cask/tree/c069add7bf974d958416e6bcc364769b378baad1/doc/cask_language_reference#stanza-order)).

Comment (vitorgalvao) [inline Casks/facebook-ios-sdk.rb]: You should have left it where it was. It was correct. My point was simply that it had no bearing to execution order, as your commit message suggested.

**Review by vitorgalvao — APPROVED** [review]

**Comment by vitorgalvao** [comment]
Thank you.

**Pull request merged**